src/targets/gpu/fuse_ck.cpp

```diff
@@ -161,10 +203,69 @@ struct find_ck_gemm
     }
 };
 
+static auto is_mul_module(module& m)
+{
+    auto is_mul =
+        match::arg(0)(match::name("mul")(match::all_of[match::inputs()](match::name("@param"))));
+    return match_instruction(m, std::prev(m.end()), is_mul).result != m.end();
+}
+
+MIGRAPHX_PRED_MATCHER(is_pointwise_scale, instruction_ref ins)
+{
+    if(ins->name() != "pointwise")
+        return false;
+    if(ins->module_inputs().size() != 1)
+        return false;
+    return is_mul_module(*ins->module_inputs().front());
+}
+
+struct find_ck_gemm_softmax_gemm
+{
+    auto matcher() const
+    {
+        auto gemm1 =
+            match::skip(match::name("contiguous"))(match::name("dot")(is_ck_gemm().bind("gemm1")));
+        auto mul     = match::name("pointwise")(match::either_arg(0, 1)(
+            match::is_constant().bind("scale"), gemm1))(is_pointwise_scale());
+        auto softmax = match::name("softmax")(match::arg(0)(mul)).bind("softmax");
+
+        return match::name("dot")(is_ck_gemm().bind("gemm2"))(match::arg(0)(softmax));
+    }
+
+    void apply(module_pass_manager& mpm, const match::matcher_result& r) const
+    {
+        std::cout << "Matched GSG" << std::endl;
```

Comment:
Remove `cout`. If you need to, 
You can also set both  `MIGRAPHX_TRACE_MATCHES=1` and `MIGRAPHX_TRACE_MATCHES_FOR="ck_gemm_softmax_gemm` to see matched instructions for just this matcher.